File: January2022/C++/Maximize_Distance_to_Closest_Person.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int ans=0,n=seats.size(),temp=0,lead=0,i =0;
        while(i<n&&seats[i]!=1){
            lead++;
            i++;
        }
        for(;i<n;i++){
            if(seats[i]==0){
                temp++;
                ans=max(ans,temp);
            }else
                temp=0;
        }
        return max((ans+1)/2,max(temp,lead));
    }
};

File: January2022/Python/maximize_distance_to_closest_person.py

# Author :  Karthik K (https://www.linkedin.com/in/karthikmunirathinam/)
# TC : O(N)
# SC : O(N)

class Solution:
    def maxDistToClosest(self, seats: List[int]) -> int:
        result = 0
        curr = 0
        for ind in range(len(seats)):
            if seats[ind] != 1:
                curr += 1
                result = max(result, (curr+1)//2)
            else:
                curr = 0
        for ind in range(len(seats)):
            if seats[ind] == 1:
                result = max(result, ind)
                break
        for ind in range(len(seats)-1, -1, -1):
            if seats[ind] == 1:
                result = max(result, len(seats)-ind-1)
                break
        return result
